src/DataGenerator.cpp

```
@@ -0,0 +1,499 @@
+        "most likely have a xml model file");
+  } else {
+    // if we have a dataset xml
+    pugi::xml_node node_alternatives =
+        doc.child("dataset").child("alternatives");
+    std::stringstream strValue;
+    int nb_alt;
+    strValue << node_alternatives.child_value();
+    strValue >> nb_alt;
+    return nb_alt;
+  }
+}
+
+Performance DataGenerator::getAlternativePerformance(std::string fileName,
+                                                     std::string alt_id) {
+  pugi::xml_document doc;
+  std::string path = "../data/" + fileName;
+  if (!doc.load_file(path.c_str()))
+    throw std::invalid_argument("Cannot open xml file, please check path");
+
+  if (DataGenerator::getXmlFileType(fileName) == "model") {
+    // if we have a model xml
+    throw std::invalid_argument("Cannot find any alternatives in xml file, "
+                                "most likely have a xml model file");
+  } else {
+
+    // creating type blocks to return Performance object
+    int nb_criteria = DataGenerator::getNumberOfCriteria(path);
+    Criteria criteria = Criteria(nb_criteria, "crit");
+    std::vector<float> alt_perf;
+
+    pugi::xml_node node_dataset = doc.child("dataset");
+    for (pugi::xml_node_iterator it = node_dataset.begin();
+         it != node_dataset.end(); ++it) {
+
+      if (it->child_value() == alt_id) {
+        // std::cout << "alternative : " << it->child_value() << std::endl;
+        pugi::xml_node alternative_node = node_dataset.child(it->name());
+
+        for (pugi::xml_node_iterator al_it = alternative_node.begin();
+             al_it != alternative_node.end(); ++al_it) {
+
+          // std::cout << "crit : " << al_it->child_value() << std::endl;
+          if (strcmp(al_it->child_value(), "assignment") != 0) {
+            float perf = atof(al_it->child_value());
+            alt_perf.push_back(perf);
+          }
+        }
+        break;
+      }
+    }
+    return Performance(alt_id, criteria, alt_perf);
```

If alt_id is not found we will return an empty Performance object, but we should probably return an error as we did not succeed in finding the alternative performance !

---

Done too